app/src/org/commcare/activities/components/FormEntryInstanceUtils.java

```diff
@@ -0,0 +1,145 @@
+package org.commcare.activities.components;
+
+import android.content.ContentUris;
+import android.content.Context;
+import android.content.Intent;
+import android.database.Cursor;
+import android.net.Uri;
+import android.util.Pair;
+import android.view.View;
+
+import org.commcare.activities.FormEntryActivity;
+import org.commcare.dalvik.R;
+import org.commcare.logging.analytics.GoogleAnalyticsFields;
+import org.commcare.logging.analytics.GoogleAnalyticsUtils;
+import org.commcare.provider.FormsProviderAPI;
+import org.commcare.provider.InstanceProviderAPI;
+import org.commcare.utils.StringUtils;
+import org.commcare.views.dialogs.DialogChoiceItem;
+import org.commcare.views.dialogs.PaneledChoiceDialog;
+
+/**
+ * @author Phillip Mates (pmates@dimagi.com)
+ */
+public class FormEntryInstanceUtils {
```

Comment:
What do you think about putting all of these methods into `FormEntryInstanceState.java`? I feel like there is a balance to strike here between making mammoth classes smaller and having an unnecessary number of small classes that don't have a clear reason for being separate entities and make it kind of cumbersome to navigate through code.